editor/editor_build_profile.cpp

```diff
@@ -0,0 +1,798 @@
+	ERR_FAIL_INDEX_V(p_build_option, BUILD_OPTION_MAX, false);
+	return build_option_disable_values[p_build_option];
+}
+
+void EditorBuildProfile::set_force_detect_classes(const String &p_classes) {
+	force_detect_classes = p_classes;
+}
+
+String EditorBuildProfile::get_force_detect_classes() const {
+	return force_detect_classes;
+}
+
+String EditorBuildProfile::get_build_option_name(BuildOption p_build_option) {
+	ERR_FAIL_INDEX_V(p_build_option, BUILD_OPTION_MAX, String());
+	const char *build_option_names[BUILD_OPTION_MAX] = {
+		TTRC("3D Engine"),
+		TTRC("2D Physics"),
+		TTRC("3D Physics"),
+		TTRC("Navigation"),
+		TTRC("XR"),
+		TTRC("RenderingDevice"),
+		TTRC("OpenGL"),
+		TTRC("Vulkan"),
+	};
+	return TTRGET(build_option_names[p_build_option]);
+}
+
+String EditorBuildProfile::get_build_option_description(BuildOption p_build_option) {
+	ERR_FAIL_INDEX_V(p_build_option, BUILD_OPTION_MAX, String());
+
+	const char *build_option_descriptions[BUILD_OPTION_MAX] = {
+		TTRC("3D Nodes as well as RenderingServer access to 3D features."),
+		TTRC("2D Physics nodes and PhysicsServer2D."),
+		TTRC("3D Physics nodes and PhysicsServer3D."),
+		TTRC("Navigation, both 2D and 3D."),
+		TTRC("XR (AR and VR)."),
+		TTRC("RenderingDevice based rendering (requires OpenGL back-end)."),
+		TTRC("OpenGL back-end (requires RenderingDevice back-end)."),
+		TTRC("Vulkan back-end of RenderingDevice."),
+	};
+
+	return TTRGET(build_option_descriptions[p_build_option]);
+}
+
+Error EditorBuildProfile::save_to_file(const String &p_path) {
+	Dictionary data;
+	data["type"] = "build_profile";
+	Array dis_classes;
+	for (const StringName &E : disabled_classes) {
+		dis_classes.push_back(String(E));
+	}
+	dis_classes.sort();
+	data["disabled_classes"] = dis_classes;
+
+	Dictionary dis_build_options;
+	for (int i = 0; i < BUILD_OPTION_MAX; i++) {
+		if (build_options_disabled[i]) {
+			dis_build_options[build_option_identifiers[i]] = build_option_disable_values[i];
+		}
+	}
+
+	data["disabled_build_options"] = dis_build_options;
+
+	if (!force_detect_classes.is_empty()) {
+		data["force_detect_classes"] = force_detect_classes;
+	}
+
+	Ref<FileAccess> f = FileAccess::open(p_path, FileAccess::WRITE);
+	ERR_FAIL_COND_V_MSG(f.is_null(), ERR_CANT_CREATE, "Cannot create file '" + p_path + "'.");
+
+	JSON json;
+	String text = json.stringify(data, "\t");
+	f->store_string(text);
+	return OK;
+}
+
+Error EditorBuildProfile::load_from_file(const String &p_path) {
+	Error err;
+	String text = FileAccess::get_file_as_string(p_path, &err);
+	if (err != OK) {
+		return err;
+	}
+
+	JSON json;
+	err = json.parse(text);
+	if (err != OK) {
+		ERR_PRINT("Error parsing '" + p_path + "' on line " + itos(json.get_error_line()) + ": " + json.get_error_message());
+		return ERR_PARSE_ERROR;
+	}
+
+	Dictionary data = json.get_data();
+
+	if (!data.has("type") || String(data["type"]) != "build_profile") {
+		ERR_PRINT("Error parsing '" + p_path + "', it's not a build_option profile.");
```

Comment:
Doesn't seem like "build_option profile" matches the tags actually used in the file. You're checking for `build_profile` above, not `build_option`.